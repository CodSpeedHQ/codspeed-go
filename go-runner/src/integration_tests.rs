use itertools::Itertools;
use rstest::rstest;
use std::path::{Path, PathBuf};
use std::sync::Mutex;
use tempfile::TempDir;

use crate::results::walltime_results::WalltimeResults;

fn setup_test_project(project_name: &str) -> anyhow::Result<TempDir> {
    let project_path = PathBuf::from(env!("CARGO_MANIFEST_DIR"))
        .join("testdata/projects")
        .join(project_name);
    println!("Project path: {project_path:?}");

    let temp_dir = TempDir::new()?;
    crate::utils::copy_dir_all(&project_path, &temp_dir)?;

    Ok(temp_dir)
}

fn assert_results_snapshots(profile_dir: &Path, project_name: &str) {
    let glob_pattern = profile_dir.join("results");
    if !glob_pattern.exists() {
        eprintln!("No results found for project: {project_name}");
        return;
    }

    let files = std::fs::read_dir(&glob_pattern)
        .expect("Failed to read results directory")
        .filter_map(Result::ok)
        .map(|entry| entry.path())
        .map(|path| {
            let file = std::fs::File::open(&path).unwrap();
            serde_json::from_reader::<_, WalltimeResults>(file).unwrap()
        })
        // Ensure we have the correct order for multiple test executables
        .sorted_by_cached_key(|r| {
            r.benchmarks
                .iter()
                .map(|b| b.metadata.name.clone())
                .sorted()
                .join(";")
        })
        .collect::<Vec<_>>();

    for (i, mut content) in files.into_iter().enumerate() {
        content
            .benchmarks
            .sort_by_cached_key(|b| b.metadata.name.clone());

        let _guard = {
            let mut settings = insta::Settings::clone_current();
            settings.set_snapshot_suffix(format!("{project_name}_{i}"));
            settings.bind_to_scope()
        };

        insta::assert_json_snapshot!(content, {
            ".creator.pid" => "[pid]",
            ".benchmarks[].stats" => "[stats]",
        });
    }
}

#[rstest]
// // #[case::caddy("caddy")]
#[case::fzf("fzf")]
#[case::opentelemetry_go("opentelemetry-go")]
#[case::golang_benchmarks("golang-benchmarks")]
#[case::zerolog("zerolog")]
#[case::zap("zap")]
#[case::hugo("hugo")]
// Currently not producing results:
#[case::fuego("fuego")]
#[case::cli_runtime("cli-runtime")]
fn test_build_and_run(#[case] project_name: &str) {
    let temp_dir = setup_test_project(project_name).unwrap();

    // Mutex to prevent concurrent tests from interfering with CODSPEED_PROFILE_FOLDER env var
    static ENV_MUTEX: Mutex<()> = Mutex::new(());
    let _env_guard = ENV_MUTEX.lock().unwrap_or_else(|e| e.into_inner());

    let profile_dir = temp_dir.path().join("profile");
    unsafe { std::env::set_var("CODSPEED_PROFILE_FOLDER", &profile_dir) };
    if let Err(error) = crate::run_benchmarks(temp_dir.path(), ".") {
        panic!("Benchmarks couldn't run: {error}");
    }

    assert_results_snapshots(&profile_dir, project_name);
}
