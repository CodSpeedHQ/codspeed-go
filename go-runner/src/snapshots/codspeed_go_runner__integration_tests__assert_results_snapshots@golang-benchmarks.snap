---
source: go-runner/src/integration_tests.rs
expression: results
---
[
  {
    "creator": {
      "name": "codspeed-go",
      "version": "[version]",
      "pid": "[pid]"
    },
    "instrument": {
      "type": "walltime"
    },
    "benchmarks": [
      {
        "name": "BenchmarkBase64decode",
        "uri": "base64/base64_test.go::BenchmarkBase64decode",
        "config": {
          "warmup_time_ns": null,
          "min_round_time_ns": null,
          "max_time_ns": null,
          "max_rounds": null
        },
        "stats": "[stats]"
      },
      {
        "name": "BenchmarkBase64regex",
        "uri": "base64/base64_test.go::BenchmarkBase64regex",
        "config": {
          "warmup_time_ns": null,
          "min_round_time_ns": null,
          "max_time_ns": null,
          "max_rounds": null
        },
        "stats": "[stats]"
      }
    ]
  },
  {
    "creator": {
      "name": "codspeed-go",
      "version": "[version]",
      "pid": "[pid]"
    },
    "instrument": {
      "type": "walltime"
    },
    "benchmarks": [
      {
        "name": "BenchmarkCompileMatch",
        "uri": "contains/contains_test.go::BenchmarkCompileMatch",
        "config": {
          "warmup_time_ns": null,
          "min_round_time_ns": null,
          "max_time_ns": null,
          "max_rounds": null
        },
        "stats": "[stats]"
      },
      {
        "name": "BenchmarkCompileMatchNot",
        "uri": "contains/contains_test.go::BenchmarkCompileMatchNot",
        "config": {
          "warmup_time_ns": null,
          "min_round_time_ns": null,
          "max_time_ns": null,
          "max_rounds": null
        },
        "stats": "[stats]"
      },
      {
        "name": "BenchmarkContains",
        "uri": "contains/contains_test.go::BenchmarkContains",
        "config": {
          "warmup_time_ns": null,
          "min_round_time_ns": null,
          "max_time_ns": null,
          "max_rounds": null
        },
        "stats": "[stats]"
      },
      {
        "name": "BenchmarkContainsBytes",
        "uri": "contains/contains_test.go::BenchmarkContainsBytes",
        "config": {
          "warmup_time_ns": null,
          "min_round_time_ns": null,
          "max_time_ns": null,
          "max_rounds": null
        },
        "stats": "[stats]"
      },
      {
        "name": "BenchmarkContainsBytesNot",
        "uri": "contains/contains_test.go::BenchmarkContainsBytesNot",
        "config": {
          "warmup_time_ns": null,
          "min_round_time_ns": null,
          "max_time_ns": null,
          "max_rounds": null
        },
        "stats": "[stats]"
      },
      {
        "name": "BenchmarkContainsMethods::Bytes.Contains",
        "uri": "contains/contains_test.go::BenchmarkContainsMethods::Bytes.Contains",
        "config": {
          "warmup_time_ns": null,
          "min_round_time_ns": null,
          "max_time_ns": null,
          "max_rounds": null
        },
        "stats": "[stats]"
      },
      {
        "name": "BenchmarkContainsMethods::RegexMatch",
        "uri": "contains/contains_test.go::BenchmarkContainsMethods::RegexMatch",
        "config": {
          "warmup_time_ns": null,
          "min_round_time_ns": null,
          "max_time_ns": null,
          "max_rounds": null
        },
        "stats": "[stats]"
      },
      {
        "name": "BenchmarkContainsMethods::RegexMatchString",
        "uri": "contains/contains_test.go::BenchmarkContainsMethods::RegexMatchString",
        "config": {
          "warmup_time_ns": null,
          "min_round_time_ns": null,
          "max_time_ns": null,
          "max_rounds": null
        },
        "stats": "[stats]"
      },
      {
        "name": "BenchmarkContainsMethods::Strings.Contains",
        "uri": "contains/contains_test.go::BenchmarkContainsMethods::Strings.Contains",
        "config": {
          "warmup_time_ns": null,
          "min_round_time_ns": null,
          "max_time_ns": null,
          "max_rounds": null
        },
        "stats": "[stats]"
      },
      {
        "name": "BenchmarkContainsNot",
        "uri": "contains/contains_test.go::BenchmarkContainsNot",
        "config": {
          "warmup_time_ns": null,
          "min_round_time_ns": null,
          "max_time_ns": null,
          "max_rounds": null
        },
        "stats": "[stats]"
      },
      {
        "name": "BenchmarkMatch",
        "uri": "contains/contains_test.go::BenchmarkMatch",
        "config": {
          "warmup_time_ns": null,
          "min_round_time_ns": null,
          "max_time_ns": null,
          "max_rounds": null
        },
        "stats": "[stats]"
      },
      {
        "name": "BenchmarkMatchNot",
        "uri": "contains/contains_test.go::BenchmarkMatchNot",
        "config": {
          "warmup_time_ns": null,
          "min_round_time_ns": null,
          "max_time_ns": null,
          "max_rounds": null
        },
        "stats": "[stats]"
      }
    ]
  },
  {
    "creator": {
      "name": "codspeed-go",
      "version": "[version]",
      "pid": "[pid]"
    },
    "instrument": {
      "type": "walltime"
    },
    "benchmarks": [
      {
        "name": "BenchmarkConcat::Buffer",
        "uri": "concat/concat_test.go::BenchmarkConcat::Buffer",
        "config": {
          "warmup_time_ns": null,
          "min_round_time_ns": null,
          "max_time_ns": null,
          "max_rounds": null
        },
        "stats": "[stats]"
      },
      {
        "name": "BenchmarkConcat::Builder",
        "uri": "concat/concat_test.go::BenchmarkConcat::Builder",
        "config": {
          "warmup_time_ns": null,
          "min_round_time_ns": null,
          "max_time_ns": null,
          "max_rounds": null
        },
        "stats": "[stats]"
      },
      {
        "name": "BenchmarkConcat::String",
        "uri": "concat/concat_test.go::BenchmarkConcat::String",
        "config": {
          "warmup_time_ns": null,
          "min_round_time_ns": null,
          "max_time_ns": null,
          "max_rounds": null
        },
        "stats": "[stats]"
      },
      {
        "name": "BenchmarkConcatBuffer",
        "uri": "concat/concat_test.go::BenchmarkConcatBuffer",
        "config": {
          "warmup_time_ns": null,
          "min_round_time_ns": null,
          "max_time_ns": null,
          "max_rounds": null
        },
        "stats": "[stats]"
      },
      {
        "name": "BenchmarkConcatBuilder",
        "uri": "concat/concat_test.go::BenchmarkConcatBuilder",
        "config": {
          "warmup_time_ns": null,
          "min_round_time_ns": null,
          "max_time_ns": null,
          "max_rounds": null
        },
        "stats": "[stats]"
      },
      {
        "name": "BenchmarkConcatString",
        "uri": "concat/concat_test.go::BenchmarkConcatString",
        "config": {
          "warmup_time_ns": null,
          "min_round_time_ns": null,
          "max_time_ns": null,
          "max_rounds": null
        },
        "stats": "[stats]"
      }
    ]
  },
  {
    "creator": {
      "name": "codspeed-go",
      "version": "[version]",
      "pid": "[pid]"
    },
    "instrument": {
      "type": "walltime"
    },
    "benchmarks": [
      {
        "name": "BenchmarkCryptoRand",
        "uri": "random/random_test.go::BenchmarkCryptoRand",
        "config": {
          "warmup_time_ns": null,
          "min_round_time_ns": null,
          "max_time_ns": null,
          "max_rounds": null
        },
        "stats": "[stats]"
      },
      {
        "name": "BenchmarkCryptoRandBytes",
        "uri": "random/random_test.go::BenchmarkCryptoRandBytes",
        "config": {
          "warmup_time_ns": null,
          "min_round_time_ns": null,
          "max_time_ns": null,
          "max_rounds": null
        },
        "stats": "[stats]"
      },
      {
        "name": "BenchmarkCryptoRandString",
        "uri": "random/random_test.go::BenchmarkCryptoRandString",
        "config": {
          "warmup_time_ns": null,
          "min_round_time_ns": null,
          "max_time_ns": null,
          "max_rounds": null
        },
        "stats": "[stats]"
      },
      {
        "name": "BenchmarkMathRand",
        "uri": "random/random_test.go::BenchmarkMathRand",
        "config": {
          "warmup_time_ns": null,
          "min_round_time_ns": null,
          "max_time_ns": null,
          "max_rounds": null
        },
        "stats": "[stats]"
      }
    ]
  },
  {
    "creator": {
      "name": "codspeed-go",
      "version": "[version]",
      "pid": "[pid]"
    },
    "instrument": {
      "type": "walltime"
    },
    "benchmarks": [
      {
        "name": "BenchmarkEmbed",
        "uri": "embed/embed_test.go::BenchmarkEmbed",
        "config": {
          "warmup_time_ns": null,
          "min_round_time_ns": null,
          "max_time_ns": null,
          "max_rounds": null
        },
        "stats": "[stats]"
      },
      {
        "name": "BenchmarkIoutilReadFile",
        "uri": "embed/embed_test.go::BenchmarkIoutilReadFile",
        "config": {
          "warmup_time_ns": null,
          "min_round_time_ns": null,
          "max_time_ns": null,
          "max_rounds": null
        },
        "stats": "[stats]"
      },
      {
        "name": "BenchmarkReadFile",
        "uri": "embed/embed_test.go::BenchmarkReadFile",
        "config": {
          "warmup_time_ns": null,
          "min_round_time_ns": null,
          "max_time_ns": null,
          "max_rounds": null
        },
        "stats": "[stats]"
      }
    ]
  },
  {
    "creator": {
      "name": "codspeed-go",
      "version": "[version]",
      "pid": "[pid]"
    },
    "instrument": {
      "type": "walltime"
    },
    "benchmarks": [
      {
        "name": "BenchmarkEqualFold",
        "uri": "caseinsensitivecompare/caseinsensitivecompare_test.go::BenchmarkEqualFold",
        "config": {
          "warmup_time_ns": null,
          "min_round_time_ns": null,
          "max_time_ns": null,
          "max_rounds": null
        },
        "stats": "[stats]"
      },
      {
        "name": "BenchmarkToLower",
        "uri": "caseinsensitivecompare/caseinsensitivecompare_test.go::BenchmarkToLower",
        "config": {
          "warmup_time_ns": null,
          "min_round_time_ns": null,
          "max_time_ns": null,
          "max_rounds": null
        },
        "stats": "[stats]"
      },
      {
        "name": "BenchmarkToUpper",
        "uri": "caseinsensitivecompare/caseinsensitivecompare_test.go::BenchmarkToUpper",
        "config": {
          "warmup_time_ns": null,
          "min_round_time_ns": null,
          "max_time_ns": null,
          "max_rounds": null
        },
        "stats": "[stats]"
      }
    ]
  },
  {
    "creator": {
      "name": "codspeed-go",
      "version": "[version]",
      "pid": "[pid]"
    },
    "instrument": {
      "type": "walltime"
    },
    "benchmarks": [
      {
        "name": "BenchmarkForMap",
        "uri": "foreach/foreach_test.go::BenchmarkForMap",
        "config": {
          "warmup_time_ns": null,
          "min_round_time_ns": null,
          "max_time_ns": null,
          "max_rounds": null
        },
        "stats": "[stats]"
      },
      {
        "name": "BenchmarkRangeMap",
        "uri": "foreach/foreach_test.go::BenchmarkRangeMap",
        "config": {
          "warmup_time_ns": null,
          "min_round_time_ns": null,
          "max_time_ns": null,
          "max_rounds": null
        },
        "stats": "[stats]"
      },
      {
        "name": "BenchmarkRangeSlice",
        "uri": "foreach/foreach_test.go::BenchmarkRangeSlice",
        "config": {
          "warmup_time_ns": null,
          "min_round_time_ns": null,
          "max_time_ns": null,
          "max_rounds": null
        },
        "stats": "[stats]"
      },
      {
        "name": "BenchmarkRangeSliceKey",
        "uri": "foreach/foreach_test.go::BenchmarkRangeSliceKey",
        "config": {
          "warmup_time_ns": null,
          "min_round_time_ns": null,
          "max_time_ns": null,
          "max_rounds": null
        },
        "stats": "[stats]"
      }
    ]
  },
  {
    "creator": {
      "name": "codspeed-go",
      "version": "[version]",
      "pid": "[pid]"
    },
    "instrument": {
      "type": "walltime"
    },
    "benchmarks": [
      {
        "name": "BenchmarkFulltextParse",
        "uri": "between/between_test.go::BenchmarkFulltextParse",
        "config": {
          "warmup_time_ns": null,
          "min_round_time_ns": null,
          "max_time_ns": null,
          "max_rounds": null
        },
        "stats": "[stats]"
      },
      {
        "name": "BenchmarkFulltextRegEx",
        "uri": "between/between_test.go::BenchmarkFulltextRegEx",
        "config": {
          "warmup_time_ns": null,
          "min_round_time_ns": null,
          "max_time_ns": null,
          "max_rounds": null
        },
        "stats": "[stats]"
      },
      {
        "name": "BenchmarkNumberParse",
        "uri": "between/between_test.go::BenchmarkNumberParse",
        "config": {
          "warmup_time_ns": null,
          "min_round_time_ns": null,
          "max_time_ns": null,
          "max_rounds": null
        },
        "stats": "[stats]"
      },
      {
        "name": "BenchmarkNumberRegEx",
        "uri": "between/between_test.go::BenchmarkNumberRegEx",
        "config": {
          "warmup_time_ns": null,
          "min_round_time_ns": null,
          "max_time_ns": null,
          "max_rounds": null
        },
        "stats": "[stats]"
      }
    ]
  },
  {
    "creator": {
      "name": "codspeed-go",
      "version": "[version]",
      "pid": "[pid]"
    },
    "instrument": {
      "type": "walltime"
    },
    "benchmarks": [
      {
        "name": "BenchmarkHTMLTemplate",
        "uri": "template/template_test.go::BenchmarkHTMLTemplate",
        "config": {
          "warmup_time_ns": null,
          "min_round_time_ns": null,
          "max_time_ns": null,
          "max_rounds": null
        },
        "stats": "[stats]"
      },
      {
        "name": "BenchmarkRegExp",
        "uri": "template/template_test.go::BenchmarkRegExp",
        "config": {
          "warmup_time_ns": null,
          "min_round_time_ns": null,
          "max_time_ns": null,
          "max_rounds": null
        },
        "stats": "[stats]"
      },
      {
        "name": "BenchmarkTextTemplate",
        "uri": "template/template_test.go::BenchmarkTextTemplate",
        "config": {
          "warmup_time_ns": null,
          "min_round_time_ns": null,
          "max_time_ns": null,
          "max_rounds": null
        },
        "stats": "[stats]"
      }
    ]
  },
  {
    "creator": {
      "name": "codspeed-go",
      "version": "[version]",
      "pid": "[pid]"
    },
    "instrument": {
      "type": "walltime"
    },
    "benchmarks": [
      {
        "name": "BenchmarkJsonMarshal",
        "uri": "json/json_test.go::BenchmarkJsonMarshal",
        "config": {
          "warmup_time_ns": null,
          "min_round_time_ns": null,
          "max_time_ns": null,
          "max_rounds": null
        },
        "stats": "[stats]"
      },
      {
        "name": "BenchmarkJsonUnmarshal",
        "uri": "json/json_test.go::BenchmarkJsonUnmarshal",
        "config": {
          "warmup_time_ns": null,
          "min_round_time_ns": null,
          "max_time_ns": null,
          "max_rounds": null
        },
        "stats": "[stats]"
      }
    ]
  },
  {
    "creator": {
      "name": "codspeed-go",
      "version": "[version]",
      "pid": "[pid]"
    },
    "instrument": {
      "type": "walltime"
    },
    "benchmarks": [
      {
        "name": "BenchmarkMapIntIndex",
        "uri": "index/index_test.go::BenchmarkMapIntIndex",
        "config": {
          "warmup_time_ns": null,
          "min_round_time_ns": null,
          "max_time_ns": null,
          "max_rounds": null
        },
        "stats": "[stats]"
      },
      {
        "name": "BenchmarkMapIntKeys",
        "uri": "index/index_test.go::BenchmarkMapIntKeys",
        "config": {
          "warmup_time_ns": null,
          "min_round_time_ns": null,
          "max_time_ns": null,
          "max_rounds": null
        },
        "stats": "[stats]"
      },
      {
        "name": "BenchmarkMapStringIndex",
        "uri": "index/index_test.go::BenchmarkMapStringIndex",
        "config": {
          "warmup_time_ns": null,
          "min_round_time_ns": null,
          "max_time_ns": null,
          "max_rounds": null
        },
        "stats": "[stats]"
      },
      {
        "name": "BenchmarkMapStringKeys",
        "uri": "index/index_test.go::BenchmarkMapStringKeys",
        "config": {
          "warmup_time_ns": null,
          "min_round_time_ns": null,
          "max_time_ns": null,
          "max_rounds": null
        },
        "stats": "[stats]"
      }
    ]
  },
  {
    "creator": {
      "name": "codspeed-go",
      "version": "[version]",
      "pid": "[pid]"
    },
    "instrument": {
      "type": "walltime"
    },
    "benchmarks": [
      {
        "name": "BenchmarkMatchString",
        "uri": "regexp/regexp_test.go::BenchmarkMatchString",
        "config": {
          "warmup_time_ns": null,
          "min_round_time_ns": null,
          "max_time_ns": null,
          "max_rounds": null
        },
        "stats": "[stats]"
      },
      {
        "name": "BenchmarkMatchStringCompiled",
        "uri": "regexp/regexp_test.go::BenchmarkMatchStringCompiled",
        "config": {
          "warmup_time_ns": null,
          "min_round_time_ns": null,
          "max_time_ns": null,
          "max_rounds": null
        },
        "stats": "[stats]"
      },
      {
        "name": "BenchmarkMatchStringGolibs",
        "uri": "regexp/regexp_test.go::BenchmarkMatchStringGolibs",
        "config": {
          "warmup_time_ns": null,
          "min_round_time_ns": null,
          "max_time_ns": null,
          "max_rounds": null
        },
        "stats": "[stats]"
      }
    ]
  },
  {
    "creator": {
      "name": "codspeed-go",
      "version": "[version]",
      "pid": "[pid]"
    },
    "instrument": {
      "type": "walltime"
    },
    "benchmarks": [
      {
        "name": "BenchmarkMathAtomicInt32",
        "uri": "math/math_test.go::BenchmarkMathAtomicInt32",
        "config": {
          "warmup_time_ns": null,
          "min_round_time_ns": null,
          "max_time_ns": null,
          "max_rounds": null
        },
        "stats": "[stats]"
      },
      {
        "name": "BenchmarkMathAtomicInt64",
        "uri": "math/math_test.go::BenchmarkMathAtomicInt64",
        "config": {
          "warmup_time_ns": null,
          "min_round_time_ns": null,
          "max_time_ns": null,
          "max_rounds": null
        },
        "stats": "[stats]"
      },
      {
        "name": "BenchmarkMathFloat32",
        "uri": "math/math_test.go::BenchmarkMathFloat32",
        "config": {
          "warmup_time_ns": null,
          "min_round_time_ns": null,
          "max_time_ns": null,
          "max_rounds": null
        },
        "stats": "[stats]"
      },
      {
        "name": "BenchmarkMathFloat64",
        "uri": "math/math_test.go::BenchmarkMathFloat64",
        "config": {
          "warmup_time_ns": null,
          "min_round_time_ns": null,
          "max_time_ns": null,
          "max_rounds": null
        },
        "stats": "[stats]"
      },
      {
        "name": "BenchmarkMathInt32",
        "uri": "math/math_test.go::BenchmarkMathInt32",
        "config": {
          "warmup_time_ns": null,
          "min_round_time_ns": null,
          "max_time_ns": null,
          "max_rounds": null
        },
        "stats": "[stats]"
      },
      {
        "name": "BenchmarkMathInt64",
        "uri": "math/math_test.go::BenchmarkMathInt64",
        "config": {
          "warmup_time_ns": null,
          "min_round_time_ns": null,
          "max_time_ns": null,
          "max_rounds": null
        },
        "stats": "[stats]"
      },
      {
        "name": "BenchmarkMathInt8",
        "uri": "math/math_test.go::BenchmarkMathInt8",
        "config": {
          "warmup_time_ns": null,
          "min_round_time_ns": null,
          "max_time_ns": null,
          "max_rounds": null
        },
        "stats": "[stats]"
      },
      {
        "name": "BenchmarkMathMutexInt",
        "uri": "math/math_test.go::BenchmarkMathMutexInt",
        "config": {
          "warmup_time_ns": null,
          "min_round_time_ns": null,
          "max_time_ns": null,
          "max_rounds": null
        },
        "stats": "[stats]"
      }
    ]
  },
  {
    "creator": {
      "name": "codspeed-go",
      "version": "[version]",
      "pid": "[pid]"
    },
    "instrument": {
      "type": "walltime"
    },
    "benchmarks": [
      {
        "name": "BenchmarkParseBool",
        "uri": "parse/parse_test.go::BenchmarkParseBool",
        "config": {
          "warmup_time_ns": null,
          "min_round_time_ns": null,
          "max_time_ns": null,
          "max_rounds": null
        },
        "stats": "[stats]"
      },
      {
        "name": "BenchmarkParseFloat",
        "uri": "parse/parse_test.go::BenchmarkParseFloat",
        "config": {
          "warmup_time_ns": null,
          "min_round_time_ns": null,
          "max_time_ns": null,
          "max_rounds": null
        },
        "stats": "[stats]"
      },
      {
        "name": "BenchmarkParseInt",
        "uri": "parse/parse_test.go::BenchmarkParseInt",
        "config": {
          "warmup_time_ns": null,
          "min_round_time_ns": null,
          "max_time_ns": null,
          "max_rounds": null
        },
        "stats": "[stats]"
      }
    ]
  }
]
